### PR TITLE
Dashboard: Update filter bar to use Design System

### LINKED ISSUE #4963: update dashboard filter styles
Update dashboard results area (right above the grid or table) to the new specs - some stuff has swapped places. https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=467%3A13745

Screen Shot 2021-02-16 at 12 50 14 PM

Can use Text, Button, Icon, Tooltip and Dropdown from Design System

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Part of updating the dashboard to use the new design system styles

## Summary
Updates this bar in the dashboard navigation:
(note: these are all the same width, I just didn't take screen shots with consistent dimensions)
**old**
Screen Shot 2021-02-17 at 2 57 36 PM
Screen Shot 2021-02-17 at 2 57 29 PM

**new**
Screen Shot 2021-02-17 at 2 48 18 PM
Screen Shot 2021-02-17 at 2 48 04 PM

Designs: https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?version-id=683762501&node-id=25%3A1312

## Relevant Technical Choices
NA

## To-do
NA

## User-facing changes
Can be seen above in summary.

## Testing Instructions
See that you can still reorder your stories as specified and toggle between grid and table views

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Follow testing instructions above

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #4963

### CODE REVIEW

File: assets/src/dashboard/constants/stories.js [hunk 1]
```diff
@@ -147,14 +147,51 @@ export const STORY_STATUSES = [
...
+        n,
+        'web-stories'
+      ),
+      n
+    ),
+  [STORY_STATUS.PUBLISHED_AND_FUTURE]: (n) =>
+    sprintf(
+      /* translators: %d: number of published stories in view */
+      _n(
+        'Viewing <strong>%d</strong> published story',
```
Comment (BrittanyIRL) [inline assets/src/dashboard/constants/stories.js:178]: nit: I'd rather avoid using `<strong>` tags here and assigning bold in css

**Review by BrittanyIRL — COMMENTED** [review]
Talked to design briefly this am and brought up the discrepancy in dropdown styles. They are ok with unifying it so there's no need to override the select button here. Just use the standard style and don't include a little label on the button (in figma it says 'info', see screenshot below). 
Screen Shot 2021-02-19 at 9 54 43 AM

File: assets/src/dashboard/constants/stories.js (hunk 1 above)
Comment (maxyinger) [reply assets/src/dashboard/constants/stories.js:178]: Are you just saying you'd rather not have there be bold in the design? or you'd prefer another way for me to assign bold styling to the substring?

Comment (BrittanyIRL) [reply assets/src/dashboard/constants/stories.js:178]: my immediate thought was how using that structure doesn't let us control the bolded weight specifically but let's leave it how you have it.

**Comment by maxyinger** [comment]
Now uses drop down as given by the DS:
Screen Shot 2021-02-22 at 2 26 05 PM

File: assets/src/dashboard/app/views/shared/bodyViewOptions.js [hunk 2]
```diff
@@ -48,28 +49,19 @@ const StorySortDropdownContainer = styled.div`
...
   display: flex;
   justify-content: center;
   align-items: center;
 `;
 
-const Label = styled.span`
-  ${TypographyPresets.Small};
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.gray500};
+const StyledLink = styled(Link)`
+  font-weight: 700;
```
Comment (BrittanyIRL) [inline assets/src/dashboard/app/views/shared/bodyViewOptions.js]: I just updated the typography presets to match figma again in this pr: https://github.com/google/web-stories-wp/pull/6484, can you remove this font-weight and we'll just let that happen once the type update's merged.

File: assets/src/design-system/components/dropDown/index.js [hunk 3]
```diff
@@ -63,7 +62,7 @@ export const DropDown = ({
   onMenuItemClick,
   options = [],
   placement = PLACEMENT.BOTTOM,
-  popupFillWidth = DEFAULT_POPUP_FILL_WIDTH,
```
Comment (BrittanyIRL) [inline assets/src/design-system/components/dropDown/index.js:66]: Nice catch here, I copied and pasted and was so concerned about maintaining use cases I didn't catch that this is a bool.

File: assets/src/dashboard/components/viewStyleBar/index.js [hunk 4]
```diff
@@ -61,20 +59,6 @@ const ToggleButton = styled.button`
...
-const GridIcon = styled(GridSVG).attrs(ICON_METRICS.VIEW_STYLE)`
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.gray500};
-  display: flex;
-  justify-content: flex-start;
-  align-items: center;
-`;
-
 export default function ViewStyleBar({ onPress, layoutStyle }) {
   return (
     <Container>
```
Comment (BrittanyIRL) [inline assets/src/dashboard/components/viewStyleBar/index.js:63]: would you mind also updating the tooltip to use the new one?

**Review by BrittanyIRL — COMMENTED** [review]
I do believe that the dropdown menu styles have been updates since last I touched the `<Menu />` component. I'll fix that separate. The work you've done to update this specific part of the view is right and doesn't need to be held up by that. 

I'm sure you know some tests need to get updated. One request - can you update the tooltip in here to the new one?

**Comment by embarks** [comment]
Screen Shot 2021-02-23 at 10 50 35 AM
Idk if this is related to your changes but I saw an issue with the z-index while testing

File: assets/src/dashboard/components/viewStyleBar/index.js [hunk 5]
```diff
@@ -61,33 +58,23 @@ const ToggleButton = styled.button`
...
-      <Tooltip content={VIEW_STYLE_LABELS[layoutStyle]} position="right">
+      <Tooltip
+        title={VIEW_STYLE_LABELS[layoutStyle]}
+        placement={TOOLTIP_PLACEMENT.BOTTOM_END}
+        hasTail
+      >
         <ToggleButton
```
Comment (samwhale) [inline assets/src/dashboard/components/viewStyleBar/index.js:69]: should we update the toggle button to use the new `Button`?

File: packages/e2e-tests/src/specs/dashboard/documentTitle.js [hunk 6]
```diff
@@ -35,7 +35,8 @@ describe('Document Title', () => {
     });
     await page.waitForTimeout(100);
 
-    await expect(page).toMatch('Viewing all templates');
+    await expect(page).toMatch('Viewing all');
+    await expect(page).toMatch('templates');
```
Comment (samwhale) [inline packages/e2e-tests/src/specs/dashboard/documentTitle.js:39]: are these supposed to be separate?

**Review by samwhale — APPROVED** [review]
Works great, nice job on fixing all the tests. Left you a couple questionies

File: assets/src/dashboard/components/viewStyleBar/index.js (hunk 5 above)
Comment (BrittanyIRL) [reply assets/src/dashboard/components/viewStyleBar/index.js:69]: It's a bit different of a toggle, it's actually just replacing the view.

File: assets/src/dashboard/karma/fixture.js [hunk 7]
```diff
@@ -193,6 +193,7 @@ export default class Fixture {
     // renders an extra container so it should be given the same size.
     container.style.width = '100%';
     container.style.height = '100%';
+    container.style.overflow = 'scroll';
```
Comment (BrittanyIRL) [inline assets/src/dashboard/karma/fixture.js:196]: 🎉 🎉

**Review by BrittanyIRL — APPROVED** [review]
This one was sneaky in its complexity thanks to those tests. Thanks for fixing them! 

Separate from this, I think we'll have a convo with design about the dropdown there and a possible alternate style but for now i'm a fan of the consistency and there's no regressions in any of the existing functionality there by using the new versions of things!

**Pull request merged**